### PR TITLE
PR gh #124 : Adding 2 corner cases for kvp profile string compare

### PR DESCRIPTION
**Pull request by kanjoe24**
Adding 2 corner cases for kvp profile string compare

### CODE REVIEW

File: tests/src/c_source/ut_test_kvp_profile.c [hunk 1]
```diff
@@ -110,6 +110,34 @@ void test_ut_kvp_profile_string(void)
...
+}
+
+void test_ut_kvp_profile_left_string_long(void)
+{
+    const char *checkField = "the beef is dead and also the cow";
+    char myString[UT_KVP_MAX_ELEMENT_SIZE];
+    UT_LOG_STEP("test_ut_kvp_profile_string");
+    UT_ASSERT_KVP_EQUAL_PROFILE_STRING( checkField, "decodeTest/checkStringDeadBeef");
+    UT_ASSERT_KVP_EQUAL_PROFILE_STRING( checkField, "decodeTest.checkStringDeadBeef");
+    UT_LOG("Note: these 2 tests are expected to fail, as the left string is longer than the profile");
```
Comment (kanjoe24) [inline tests/src/c_source/ut_test_kvp_profile.c:134]: Added a note for these cases, as its logged as failure by the framework.

**Comment by kanjoe24** [comment]
Logs after changes:

```shell
Enter number of suite to select (1-3) : 2
Suite 'ut-kvp - assert testing yaml ' selected.

***************** CUNIT CONSOLE - SUITE MENU ***************************
(R)un (S)elect (L)ist (A)ctivate (F)ailures (U)p (O)ptions (H)elp (Q)uit
Enter command: r

2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   636 : Running Suite : ut-kvp - assert testing yaml 
2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint8'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,    43 : test_ut_kvp_profile_uint8 - start
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,    51 : test_ut_kvp_profile_uint8 - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint8'
2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint16'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,    58 : test_ut_kvp_profile_uint16 - start
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,    66 : test_ut_kvp_profile_uint16 - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint16'
2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint32'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,    73 : test_ut_kvp_profile_uint32 - start
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,    81 : test_ut_kvp_profile_uint32 - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint32'
2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile uint64'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,    88 : test_ut_kvp_profile_uint64
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,    96 : test_ut_kvp_profile_uint64 - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile uint64'
2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile string'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   103 : test_ut_kvp_profile_string
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   110 : test_ut_kvp_profile_string - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile string'
2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile bool'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   142 : test_ut_kvp_profile_bool
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   150 : test_ut_kvp_profile_bool - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile bool'
2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile list count'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   158 : test_ut_kvp_profile_list_count
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   165 : test_ut_kvp_profile_list_count - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile list count'
2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile right string long'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   117 : test_ut_kvp_profile_string
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   123 : test_ut_kvp_profile_string - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile right string long'
2025-04-25-13:26:54, LOG   , ut_console.c,   633 : 

2025-04-25-13:26:54, LOG   , ut_console.c,   639 :      Running Test : 'kvp profile left string long'
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   130 : test_ut_kvp_profile_string
2025-04-25-13:26:54, STEP  , ut_test_kvp_profile.c,   136 : test_ut_kvp_profile_string - end
2025-04-25-13:26:54, LOG   , ut_console.c,   661 :      Test Complete : 'kvp profile left string long'

Run Summary:    Type  Total    Ran Passed Failed Inactive
              suites      3      1    n/a      0        0
               tests     19      9      9      0        0
             asserts     38     38     38      0      n/a

Elapsed time =    0.003 seconds


```

**Review by Ulrond — APPROVED** [review]

**Comment by kanjoe24** [comment]
Release scripts results below:

```shell
./release-test-script-ut-core.sh -t feature/gh124-add-cornercase-string-compare
UT_CORE_BRANCH_NAME = feature/gh124-add-cornercase-string-compare
UT_CONTROL_BRANCH_NAME = 
UT_CONTROL_BRANCH_NAME is empty
==================== RESULTS FOR C VARIANT ====================
==========================================================
RESULTS for ubuntu 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
hal_test does not exist. PASS 
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/linux/openssl/lib/libssl.a exists. PASS 
framework/ut-control/host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
hal_test does not exist. PASS 
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
hal_test does not exist. PASS 
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
CURL static lib does not exist. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
hal_test does not exist. PASS 
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for kirkstone_arm 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
hal_test does not exist. PASS 
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for kirkstone_linux 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
hal_test does not exist. PASS 
tests/build/bin/ut-test exists. PASS
==========================================================
==================== RESULTS FOR CPP VARIANT ====================
==========================================================
RESULTS for ubuntu 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
build/bin/hal_test exists. PASS
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/linux/openssl/lib/libssl.a exists. PASS 
framework/ut-control/host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
build/bin/hal_test exists. PASS
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
build/arm/cpp_libs/lib/libgtest.a exists. PASS
build/arm/cpp_libs/lib/libgtest_main.a exists. PASS
build/bin/hal_test exists. PASS
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
CURL static lib does not exist. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
GTEST static lib does not exist. FAIL 
GTEST main static lib does not exist. FAIL 
UT_CORE_BIN does not exist. FAIL 
UT_CORE_TEST_BIN does not exist. FAIL 
==========================================================
==========================================================
RESULTS for kirkstone_arm 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
build/arm/cpp_libs/lib/libgtest.a exists. PASS
build/arm/cpp_libs/lib/libgtest_main.a exists. PASS
build/bin/hal_test exists. PASS
tests/build/bin/ut-test exists. PASS
==========================================================
==========================================================
RESULTS for kirkstone_linux 
On the branch feature/gh124-add-cornercase-string-compare
        . PASS
framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
build/linux/cpp_libs/lib/libgtest.a exists. PASS
build/linux/cpp_libs/lib/libgtest_main.a exists. PASS
build/bin/hal_test exists. PASS
tests/build/bin/ut-test exists. PASS
==========================================================
jpn323@janus ~/workspace/ut-core/scripts (feature/gh124-add-cornercase-string-compare)$
``` 

Th failure case in CPP for dunfell-linux will get addressed once the PR: https://github.com/rdkcentral/ut-control/pull/72 is released.

```shell
./release-test-script-platform.sh -t feature/gh124-add-cornercase-string-compare
UT_CORE_BRANCH_NAME = feature/gh124-add-cornercase-string-compare
UT_CONTROL_BRANCH_NAME = 
UT_CONTROL_BRANCH_NAME is empty
==========================================================
RESULTS for ubuntu 
On the branch feature/gh124-add-cornercase-string-compare
            . PASS
/home/jpn323/workspace/ut-core/scripts/rdk-halif-hdmi_cec-ubuntu
ut/bin/hal_test_RCECHal exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for VM-SYNC 
On the branch feature/gh124-add-cornercase-string-compare
            . PASS
/home/jpn323/workspace/ut-core/scripts/rdk-halif-hdmi_cec-VM-SYNC_linux
ut/bin/hal_test_RCECHal exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/linux/openssl/lib/libssl.a exists. PASS 
ut/ut-core/framework/ut-control/host-tools/CMake-3.30.0/build/bin/cmake exists. PASS 
==========================================================
==========================================================
RESULTS for dunfell_arm 
On the branch feature/gh124-add-cornercase-string-compare
            . PASS
/home/jpn323/workspace/ut-core/scripts/rdk-halif-hdmi_cec-dunfell_arm
ut/bin/hal_test_RCECHal exists. PASS
ut/ut-core/framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for dunfell_linux 
On the branch feature/gh124-add-cornercase-string-compare
            . PASS
/home/jpn323/workspace/ut-core/scripts/rdk-halif-hdmi_cec-dunfell_linux
ut/bin/hal_test_RCECHal exists. PASS
CURL static lib does not exist. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for kirkstone_arm 
On the branch feature/gh124-add-cornercase-string-compare
            . PASS
/home/jpn323/workspace/ut-core/scripts/rdk-halif-hdmi_cec-kirkstone_arm
ut/bin/hal_test_RCECHal exists. PASS
ut/ut-core/framework/ut-control/build/arm/curl/lib/libcurl.a exists. PASS
ut/ut-core/framework/ut-control/build/arm/openssl/lib/libssl.a exists. PASS 
CMake host binary does not exist. PASS 
==========================================================
==========================================================
RESULTS for kirkstone_linux 
On the branch feature/gh124-add-cornercase-string-compare
            . PASS
/home/jpn323/workspace/ut-core/scripts/rdk-halif-hdmi_cec-kirkstone_linux
ut/bin/hal_test_RCECHal exists. PASS
ut/ut-core/framework/ut-control/build/linux/curl/lib/libcurl.a exists. PASS
Openssl static lib does not exist. PASS 
CMake host binary does not exist. PASS 
==========================================================
jpn323@janus ~/workspace/ut-core/scripts (feature/gh124-add-cornercase-string-compare)$ 
```
